plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2275,6 +2314,102 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
                 computedStatistics);
     }
 
+    private void generateMissingStatistics(ConnectorSession session, DeltaLakeTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
+    {
+        Map<String, AddFileEntry> addFileEntriesWithNoStats = metastore.getValidDataFiles(tableHandle.getSchemaTableName(), session).stream()
+                .filter(addFileEntry -> addFileEntry.getStats().isEmpty()
+                        || addFileEntry.getStats().get().getMaxValues().isEmpty()
+                        || addFileEntry.getStats().get().getMinValues().isEmpty()
+                        || addFileEntry.getStats().get().getNullCount().isEmpty())
+                .collect(Collectors.toUnmodifiableMap(AddFileEntry::getPath, Function.identity()));
+
+        if (addFileEntriesWithNoStats.isEmpty()) {
+            return;
+        }
+
+        Map<String, ComputedStatistics> groupedStatistics = Statistics.createComputedStatisticsToPartitionMap(computedStatistics, ImmutableList.of(PATH_COLUMN_NAME), ImmutableMap.of(PATH_COLUMN_NAME, VARCHAR))
+                .entrySet().stream()
+                .map(s -> Map.entry(s.getKey().get(0), s.getValue())) // Grouping by only one value
+                .map(s -> Map.entry(getFileName(s.getKey()), s.getValue()))// Convert path to file name
```

Comment:
Do we require two `.map`? can't we use 

`.map(stats -> Map.entry(getFileName(stats.getKey().get(0)), stats.getValue()))`